[15-min fix] Show the navigation options regardless of screen size

fdoxyz opened the pull request:
<!--
     For Work In Progress Pull Requests, please use the Draft PR feature,
     see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.

     For a timely review/response, please avoid force-pushing additional
     commits if your PR already received reviews or comments.

     Before submitting a Pull Request, please ensure you've done the following:
     - 📖 Read the Forem Contributing Guide: https://github.com/forem/forem/blob/main/CONTRIBUTING.md#create-a-pull-request.
     - 📖 Read the Forem Code of Conduct: https://github.com/forem/forem/blob/main/CODE_OF_CONDUCT.md.
     - 👷‍♀️ Create small PRs. In most cases this will be possible.
     - ✅ Provide tests for your changes.
     - 📝 Use descriptive commit messages.
     - 📗 Update any related documentation and include any relevant screenshots.

     NOTE: Pull Requests from forked repositories will need to be reviewed by
     a Forem Team member before any CI builds will run. Once your PR is approved
     with a `/ci` reply to the PR, it will be allowed to run subsequent builds without
     manual approval.
-->

## What type of PR is this? (check all applicable)

- [x] Bug Fix

## Description

This is not the final solution to be implemented for small screen navigation, but instead is a quick fix so the navigation options are available in the mean time the definitive solution is implemented.

## Related Tickets & Documents

- #13826
- https://github.com/forem/rfcs/pull/264

## QA Instructions, Screenshots, Recordings

1. Boot app locally
2. Browse the admin area with a mobile device (or alternatively with the dev tools mobile screen option enabled)

**In bigger screens (things remain the same):**

<img width="1025" alt="Screen Shot 2021-07-23 at 11 23 47" src="https://user-images.githubusercontent.com/6045239/126819076-35881c99-17ed-4fc6-b247-f11ce101d638.png">

**In small screens the navigation must appear visible:** 

<img width="1023" alt="Screen Shot 2021-07-23 at 11 23 54" src="https://user-images.githubusercontent.com/6045239/126819142-b3befb42-c466-4bb8-892c-29e329c18961.png">

### UI accessibility concerns?

None - No changes have been made to the page

## Added/updated tests?

- [x] No, and this is why: Nothing has changed other than making the navigation visible on small screens

## [Forem core team only] How will this change be communicated?

- [x] This change does not need to be communicated, and this is why not: Small fix and the navigation will be refactored when https://github.com/forem/rfcs/pull/264 is implemented

## [optional] What gif best describes this PR or how it makes you feel?

![surprise](https://media.giphy.com/media/TGqRxknWgUFypgTD2l/giphy.gif)

fdoxyz commented on app/views/layouts/admin.html.erb:
```diff
@@ -61,6 +61,12 @@
               </a>
             </li>
             <%= render "admin/shared/nested_sidebar", menu_items: AdminMenu.navigation_items %>
+            <li class="block s:hidden">
```
Small screens don't show the top-right buttons that allow creators to go back to the Forem's homepage. This is why this option is added only for small screens, so they are able to navigate back to the site.

maestromac reviewed (approved):
Tested locally. LGTM!

joshpuetz reviewed (approved):
Tested locally, in iOS simulator Safari, and iOS Forem app 1.0.4 (49): all look great!

One non blocking suggestion: the leading padding of the section headers is very small and could stand to be larger!
<img width="157" alt="CleanShot 2021-07-23 at 14 37 18@2x" src="https://user-images.githubusercontent.com/37842/126832850-31d2c70e-77f8-401f-81aa-e12028464324.png">

Ridhwana reviewed (commented):
This is great, thanks @fdoxyz!

fdoxyz commented:
Thanks everyone for the reviews! I added a small fix to the link that takes admins back to the Forem home page, nothing major 🙂 

> One non blocking suggestion: the leading padding of the section headers is very small and could stand to be larger!

I think this padding issue is a bit more widespread than just the headers @joshpuetz. Every header (from all admin sections) has this problem and all the content is flush to each side. It makes cards feel a bit weird because of their rounded corners (i.e. the main feed cards do have some padding/margin in them).

<img width="374" alt="Screen Shot 2021-07-23 at 14 14 35" src="https://user-images.githubusercontent.com/6045239/126836463-9c625729-8591-4bfb-91e8-07bdc9aa802c.png">

At first I added a very simple `p-2` to the `main` element of the admin section, and this kind of helped, but I think there's a bunch of "small tweaks" that could be bundled together and improve the experience in smaller screens. I'll put those together in a separate PR so we can get some eyes on those changes more specifically instead.

joshpuetz commented:
Cool, thanks for trying @fdoxyz, and I agree that this PR shouldn't be held up by those formatting oddities for now
